### PR TITLE
PRL-3874 - Upload your application

### PR DESCRIPTION
**Pull request by hcarslaw**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the keep-helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/application-within-proceedings/upload-your-application/content.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { AWPApplicationType, YesOrNo } from '../../../app/case/definition';
+import { TranslationFn } from '../../../app/controller/GetController';
+import { FormContent } from '../../../app/form/Form';
+import { isFieldFilledIn } from '../../../app/form/validation';
+import { getCasePartyType } from '../../../steps/prl-cases/dashboard/utils';
+import { getApplicationDetails } from '../utils';
+
+export const en = (applicationType?: AWPApplicationType): object => ({
+  title: 'Upload your application',
```
Comment (vivek-sekhar) [inline src/main/steps/application-within-proceedings/upload-your-application/content.ts:9]: ideally en & cy shd not be functions. please handle dynamic contents in generatecontent method.

File: src/main/steps/applicationWithinProceedingsSequence.ts [hunk 2]
```diff
@@ -14,7 +15,20 @@ export const applicationWithinProceedingsSequence: Step[] = [
   {
     url: APPLICATION_WITHIN_PROCEEDINGS_GUIDANCE,
     showInSection: Sections.ApplicationWithinProceedings,
-    getNextStep: () => APPLICATION_WITHIN_PROCEEDINGS_GUIDANCE,
+    getNextStep: (_userCase, req) =>
```
Comment (vivek-sekhar) [inline src/main/steps/applicationWithinProceedingsSequence.ts:18]: replace _usercase to caseData

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/applicationWithinProceedingsSequence.ts (hunk 2 above)
Comment (hcarslaw) [reply src/main/steps/applicationWithinProceedingsSequence.ts]: Changed to caseData

File: src/main/steps/application-within-proceedings/upload-your-application/content.ts (hunk 1 above)
Comment (hcarslaw) [reply src/main/steps/application-within-proceedings/upload-your-application/content.ts:10]: en and cy no longer functions and using generateContent method to handle dynamic content now

File: src/main/steps/application-within-proceedings/upload-your-application/content.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  return {
+    ...translations,
+    form,
+    fillForm: interpolate(translations.fillForm, { applicationType: applicationDetails!.applicationType }),
+    alreadyCompleted: interpolate(translations.alreadyCompleted, {
+      applicationType: applicationDetails!.applicationType,
+    }),
+    caption: applicationDetails?.reasonText,
+    errors: {
```
Comment (vivek-sekhar) [inline src/main/steps/application-within-proceedings/upload-your-application/content.ts:98]: You cannot simply override error object. Use object destructing to extend.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**